seismicpro/src/utils.py

```diff
@@ -792,3 +793,35 @@ def transform_to_fixed_width_columns(path, path_save=None, n_spaces=8, max_len=(
             if path_save:
                 return
             shutil.copyfile(write_file.name, path)
+
+def create_args(call, skip_first=False, **kwargs):
+    """ Constructing tuple with positional arguments to callable ``call`` based on
+    ``kwargs`` and ``call``'s defaults.
+
+    Parameters
+    ----------
+    call : callable
+        Function to create positional arguments for.
+    skip_first : bool
+        If True, created tuple won't contain value for frist argument.
+        if False, created tuple will contain value for first argument.
+    kwargs : dict
+        Keyword arguments to function ``call``.
+
+    Returns
+    -------
+        : tuple
+        Positional arguments to ``call``.
+    """
+    params = inspect.signature(call).parameters
+    args = list(np.zeros(len(params.keys()) - skip_first))
```

Comment:
```suggestion
    args = [kwargs.get(name, param.default) for name, param in params.items()]
```
Let's do this with the following checks empty parameters and `skip_first` flag. Should be much readable.